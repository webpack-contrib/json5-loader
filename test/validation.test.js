import loader from '../src';

it('validation', async () => {
  const json5 = '{foo: "bar"}';
  const emitError = jest.fn();

  expect(() => {
    loader.call(
      {
        emitError,
      },
      json5
    );
  }).not.toThrow();

  expect(() => {
    loader.call(
      {
        query: {
          foo: 'bar',
        },
        emitError,
      },
      json5
    );
  }).toThrowErrorMatchingSnapshot();
});
