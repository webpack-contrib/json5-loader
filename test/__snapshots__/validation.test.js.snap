// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validation 1`] = `
"JSON5 Loader Invalid Options

options should NOT have additional properties
"
`;
